### PR TITLE
Gamelab execution speed (under experiment: showDebugSlider)

### PR DESCRIPTION
**Pull request by cpirich**
*If you're interested, please chat with @poorvasingal for more context on why this is going in while incomplete and under an experiment flag*

* When `?enableExperiments=showDebugSlider` is on the URL, gamelab will now show a debug speed slider, which currently has only two modes: full speed and slow speed
* In full speed (slider all the way to the right), which is the default, everything behaves normally
* In slow speed, there are 4 notable changes that are going to be tested with the curriculum team:
  1. The interpreter execution speed is no longer "max speed", so code highlighting will occur
  2. The period of time in between `executeInterpreter()` calls is slowed from `1ms` to `100ms`
  3. The `p5` `frameRate` is set to 1 frame per second
  4. The `p5.play` debug sprite colliders are drawn after every call to `executeInterpreter()`, so you sort of see sprites "move" before you call `drawSprites()` at the end of the `draw()` loop. (Note that we don't currently erase the debug collider rectangles, so you will see "collider rectangle trails" as the sprites move around) - This would definitely need to be cleaned up, but for now, we just want to evaluate the concept.

### CODE REVIEW

File: apps/src/gamelab/GameLab.js [hunk 1]
```diff
@@ -366,6 +370,14 @@ GameLab.prototype.onIsDebuggingSpritesChange = function (isDebuggingSprites) {
   this.gameLabP5.debugSprites(isDebuggingSprites);
 };
 
+GameLab.prototype.onStepSpeedChange = function (stepSpeed) {
+  this.gameLabP5.changeStepSpeed(stepSpeed);
+
+  if (this.pauseTickTimer()) {
+    this.startTickTimer();
```
Comment (islemaster) [inline apps/src/gamelab/GameLab.js]: This took me a while to interpret.  A couple of things that might make this easier to read:

1. `pauseTickTimer` isn't really an accurate description - `stopTickTimer` or `clearTickTimer` might be clearer.
2. Documenting the return value of `pauseTickTimer` would help - and it's a little surprising that method has a return value at all.
3. Doesn't this basically vary on whether we're currently running or not?  Might read better:
   ```js
   if (this.isRunning()) {
     this.clearTickTimer();
     this.startTickTimer();
   }
   ```

File: apps/src/gamelab/GameLabP5.js [hunk 2]
```diff
@@ -451,13 +452,40 @@ GameLabP5.prototype.registerP5EventHandler = function (eventName, handler) {
   this.p5[eventName] = handler;
 };
 
+GameLabP5.prototype.setP5FrameRate = function () {
+  if (!this.p5) {
+    return;
+  }
+  if (this.stepSpeed < 1) {
+    // TODO: properly handle overriding frameRate
```
Comment (islemaster) [inline apps/src/gamelab/GameLabP5.js]: Can you clarify what you mean with this TODO?

File: apps/src/gamelab/GameLabP5.js [hunk 3]
```diff
@@ -451,13 +452,40 @@ GameLabP5.prototype.registerP5EventHandler = function (eventName, handler) {
   this.p5[eventName] = handler;
 };
 
+GameLabP5.prototype.setP5FrameRate = function () {
```
Comment (islemaster) [inline apps/src/gamelab/GameLabP5.js]: Should this method be "private"? (at least annotated as such?)

File: apps/src/gamelab/GameLabP5.js [hunk 4]
```diff
@@ -451,13 +452,40 @@ GameLabP5.prototype.registerP5EventHandler = function (eventName, handler) {
...
+  if (!this.p5) {
+    return;
+  }
+  if (this.stepSpeed < 1) {
+    // TODO: properly handle overriding frameRate
+    this.prevFrameRate = this.p5.frameRate();
+    this.p5.frameRate(1);
+  } else {
+    this.p5.frameRate(this.prevFrameRate || 30);
```
Comment (islemaster) [inline apps/src/gamelab/GameLabP5.js]: Where were we setting 30 as the default framerate before?  Should this 30 be deduplicated with that one?

File: apps/src/gamelab/GameLabP5.js (hunk 3 above)
Comment (cpirich) [reply apps/src/gamelab/GameLabP5.js]: 👍

File: apps/src/gamelab/GameLabP5.js (hunk 2 above)
Comment (cpirich) [reply apps/src/gamelab/GameLabP5.js]: 👍 
Updating the comment to read:
```
      // TODO: properly handle overriding frameRate (this implementation doesn't
      // account for any calls to frameRate() that occur while we are in the
      // slow mode - we'll need to patch p5 to capture those and update
      // this.prevFrameRate)
```

File: apps/src/gamelab/GameLabP5.js (hunk 4 above)
Comment (cpirich) [reply apps/src/gamelab/GameLabP5.js]: 👍

File: apps/src/gamelab/GameLab.js (hunk 1 above)
Comment (cpirich) [reply apps/src/gamelab/GameLab.js]: 👍 
Renamed function
Removed return value
Added `isTickTimerRunning()` function
Changed to:
```
  if (this.isTickTimerRunning()) {
    this.stopTickTimer();
    this.startTickTimer();
  }
```

**Pull request merged**